cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java

```diff
@@ -179,9 +229,11 @@ private DatasetMeta getFromMds(Id.DatasetInstance instance) throws Exception {
    * @throws DatasetTypeNotFoundException if the dataset type was not found
    * @throws Exception if something went wrong
    */
-  public void create(String namespaceId, String name, DatasetInstanceConfiguration props) throws Exception {
+  void create(String namespaceId, String name, DatasetInstanceConfiguration props) throws Exception {
     // Throws NamespaceNotFoundException if the namespace does not exist
     Id.Namespace namespace = ConversionHelpers.toNamespaceId(namespaceId);
+    Principal principal = SecurityRequestContext.toPrincipal();
```

Comment:
This class is a Service, so there is no restriction that it can only be used from the REST endpoint. E.g. currently DS is running inside Master, hence for all dataset service operations initiated from Master, it shouldn't need to go through the REST endpoint at all (through proper Guice binding). Although it's not the case right now, but we should make this class do the right thing.